.happy/terraform/modules/schema_migration/main.tf

```diff
@@ -0,0 +1,321 @@
+                        },
+                        {
+                          "Name": "DATASET_ID",
+                          "Value.$": "$.dataset_id"
+                        },
+                        {
+                          "Name": "DATASET_VERSION_ID",
+                          "Value.$": "$.dataset_version_id"
+                        },
+                        {
+                          "Name": "COLLECTION_ID",
+                          "Value.$": "$.collection_id"
+                        },
+                        {
+                            "Name": "TASK_TOKEN",
+                            "Value.$": "$$.Task.Token"
+                        }
+                      ]
+                    }
+                  },
+                  "Next": "StepFunctionsStartExecution"
+                },
+                "StepFunctionsStartExecution": {
+                  "Type": "Task",
+                  "Resource": "arn:aws:states:::states:startExecution.sync:2",
+                  "Parameters": {
+                    "StateMachineArn": "arn:aws:states:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:stateMachine:dp-${var.deployment_stage}-${var.custom_stack_name}-sfn",
+                    "Input": {
+                      "AWS_STEP_FUNCTIONS_STARTED_BY_EXECUTION_ID.$": "$$.Execution.Id",
+                      "url.$": "$.url",
+                      "dataset_id.$": "$.dataset_version_id",
+                      "collection_id.$": "$.collection_id"
+                    }
+                  },
+                  "End": true
+                }
+              }
+            },
+            "ItemsPath": "$",
+            "Next": "CollectionPublish",
+            "MaxConcurrency": 40
+          },
+          "CollectionPublish": {
+            "Type": "Task",
+            "Resource": "arn:aws:states:::batch:submitJob.sync",
+            "Parameters": {
+              "JobDefinition": "${resource.aws_batch_job_definition.schema_migrations.arn}",
+              "JobName": "collection_publish",
+              "JobQueue": "${var.job_queue_arn}",
+              "Timeout": {
+                "AttemptDurationSeconds": 600
+              },
+              "ContainerOverrides": {
+                "ResourceRequirements": [
+                  {
+                    "Type": "VCPU",
+                    "Value": "4"
+                  },
+                  {
+                    "Type": "MEMORY",
+                    "Value": "2048"
+                  }
+                ],
+                "Environment": [
+                  {
+                    "Name": "STEP_NAME",
+                    "Value": "collection_publish"
+                  },
+                  {
+                    "Name": "MIGRATE",
+                    "Value": "True"
+                  },
+                  {
+                    "Name": "COLLECTION_ID",
+                    "Value.$": "$[0].Input.collection_id"
```

Comment:
Here, the previous Map steps returns an array of datasets. I am just grabbing the first element since they're all from the same collection.